internal/terraform/resource.go

```diff
@@ -28,9 +28,29 @@ type Resource struct {
 	Version        types.String `json:"version" toml:"version" xml:"version" yaml:"version"`
 }
 
-// FullType returns full name of the type of the resource, including the provider name
-func (r *Resource) FullType() string {
-	return r.ProviderName + "_" + r.Type
+/*
+Spec Returns the resource spec addresses a specific resource in the config.
+It takes the form: resource_type.resource_name[resource index]
+
+For more details, see:
+https://www.terraform.io/docs/cli/state/resource-addressing.html#resource-spec
+*/
+func (r *Resource) Spec() string {
+	return r.ProviderName + "_" + r.Type + "." + r.Name
+}
+
+// GetMode returns normalized resource type as "resource" or "data source"
+func (r *Resource) GetMode() string {
+	mode := ""
```

Comment:
There's no need to define a variable here, on each case you can just return the value.